Sources/TuistCore/MetadataProviders/FrameworkMetadataProvider.swift

```diff
@@ -35,12 +77,38 @@ public final class FrameworkMetadataProvider: PrecompiledMetadataProvider, Frame
     }
 
     public func product(frameworkPath: AbsolutePath) throws -> Product {
-        let binaryPath = FrameworkNode.binaryPath(frameworkPath: frameworkPath)
+        let binaryPath = self.binaryPath(frameworkPath: frameworkPath)
         switch try linking(binaryPath: binaryPath) {
         case .dynamic:
             return .framework
         case .static:
             return .staticFramework
         }
     }
+
+    private func binaryPath(frameworkPath: AbsolutePath) -> AbsolutePath {
+        frameworkPath.appending(component: frameworkPath.basenameWithoutExt)
+    }
+}
+
+// MARK: - Error
+
+enum FrameworkMetadataProviderError: FatalError, Equatable {
```

Comment:
Nitpick: We've been placing the errors at the top of the file so I'd do the same here to be consistent.